src/Worker.Extensions.DurableTask/Exceptions/DurableSerializationException.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using Microsoft.DurableTask.Protobuf;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask.Exceptions;
+
+internal class DurableSerializationException : Exception
+{
+    private Exception FromException;
+
+    // We set the base class properties of this exception to the same as the parent, 
+    // so that methods in the worker after this can still (typically) access the same information vs w/o
+    // this exception type. 
+    internal DurableSerializationException(Exception fromException) : base(fromException.Message, fromException.InnerException)
+    {
+        this.FromException = fromException;
+    }
+
+    public override string ToString()
+    {
+        TaskFailureDetails? failureDetails = ExceptionToTaskFailureDetailsRecursive(this.FromException);
+        return JsonConvert.SerializeObject(failureDetails);
+    }
+
+    public override string? StackTrace => this.FromException.StackTrace;
+
+    private static TaskFailureDetails? ExceptionToTaskFailureDetailsRecursive(Exception? fromException)
```

Comment:
Do we have a shared version of this we can use? Or is it in some inaccessible place, so we have to copy it? I'm a tiny bit worried about keeping multiple copies up-to-date if we decide to add richer information to `TaskFailureDetails` (which is very likely).